services/apps/portal/src/parcel/parcel.controller.ts

```diff
@@ -0,0 +1,41 @@
+import { ServiceNotFoundException } from '@app/common/exceptions/base.exception';
+import {
+  BadRequestException,
+  Controller,
+  Get,
+  Param,
+  UseGuards,
+} from '@nestjs/common';
+import { AuthGuard } from '../common/authorization/auth-guard.service';
+import { ParcelLookupDto } from './parcel.dto';
+import { ParcelService } from './parcel.service';
+
+@Controller('parcel')
+@UseGuards(AuthGuard)
+export class ParcelController {
+  constructor(private parcelService: ParcelService) {}
+
+  @Get('search/:number')
+  async searchByPidPin(
+    @Param('number') number: string,
+  ): Promise<ParcelLookupDto | undefined> {
```

Comment:
I wonder if it possible to use @Query to perform this kind of validation on nestJS level

Reply:
Done! You can also do this with Param, just need to use a Dto